src/source-control/GitSourceControl.ts

```diff
@@ -387,16 +386,13 @@ export class GitSourceControl implements SourceControl {
         if (!branch) {
           branch = createSttackBranch();
           await repo.createBranch(branch, commit.hash, true);
-          this.repo.commits = produce(
-            this.repo.commits,
-            (draftCommitHashMap) => {
-              const commitToUpdate = draftCommitHashMap.get(commit.hash)!;
-              commitToUpdate.refNames = Array.from(
-                new Set([...commitToUpdate.refNames, branch!]),
-              );
-              draftCommitHashMap.set(commit.hash, commitToUpdate);
-            },
-          );
+          this.repo = produce(this.repo, (draftRepo) => {
+            const commitToUpdate = draftRepo.commits.get(commit.hash)!;
+            commitToUpdate.refNames = Array.from(
+              new Set([...commitToUpdate.refNames, branch]),
```

Comment:
I haven't made any changes here, I think I just changed this because it was mutating repo.

But looking at this, shouldn't this be fine to force unwrap using `!`, as we are ensuring the branch is set inside the if condition at line `487`?

Also, we do create a branch with the refName on line `488`, so I think it makes sense to update our repo (By our repo, I mean `this.repo`) by updating the refNames that point to this particular commit. 

cc @manyaagarwal - I think you wanted this function, so you probably could also look into this

Reply:
Oh I was actually trying to highlight a bigger issue. It looks like `branch` is a branch name, and we're adding that into an array of ref names. Should the branch be transformed into a ref name before adding it into that array?

Reply:
Yeah I'm pretty sure it needs to be transformed into a ref name. This doesn't look right (screenshot from #56, but the cause is unlikely to be from the new changes):

![image](https://user-images.githubusercontent.com/12784593/90326744-91232000-dfbe-11ea-89ee-940635b90077.png)

Steps I did to cause that:

1. Check out the `feat` branch, which was pointing at commit C.
1. Add a new commit, D.
1. Launch Stack Attack
1. Navigate to commit C
1. Press `s` to create PRs. The new branch name shows up in the UI, alongside all the other ref names, but the new branch should show up as a ref name.

Reply:
Looks good to me, should I go ahead and make this change?

Reply:
Yes please!

Reply:
Should be fixed now, can you check?

Reply:
Adding a screenshot here, for context in the future

<img width="946" alt="Screenshot 2020-08-16 at 1 30 56 PM" src="https://user-images.githubusercontent.com/31125345/90329753-fd088700-dfc4-11ea-9b22-7555dc5e6ffa.png">